tests/metal/texture-sampler-less.slang

```diff
@@ -0,0 +1,558 @@
+//TEST:SIMPLE(filecheck=METAL): -stage compute -entry computeMain -target metal -DEMIT_SOURCE
+//TEST:SIMPLE(filecheck=METALLIB): -stage compute -entry computeMain -target metallib -DEMIT_SOURCE -DMETALLIB
+//TEST:SIMPLE(filecheck=HLSL): -stage compute -entry computeMain -target hlsl -DEMIT_SOURCE
+
+// It appears that when kIROp_CombinedTextureSamplerGetSampler is used,
+// "$1" refers to a sampler not the first parameter of the function.
+// This causes to emit an invalid code when the implementation is shared
+// between the combined and not-combined.
+// The following code,
+//   void GetDimensions(out float width) { __intrinsic_asm "*$1 = $0.get_width(0)"; }
+// gets emitted as following,
+//   (*(kernelContext_0->t1D_sampler_0) = (kernelContext_0->t1D_texture_0).get_width(0));
+// when it is expected to be,
+//   (*&width = (kernelContext_0->t1D_texture_0).get_width(0));
+//#define TEST_WHEN_DEFAULT_SAMPLER_WORKS_PROPERLY
+
+//TEST_INPUT: ubuffer(data=[0], stride=4):out,name outputBuffer
+RWStructuredBuffer<int> outputBuffer;
+
+// TODO: `SamplerXD` cannot be setup with TEST_INPUT method.
+
+Sampler1D<float> t1D;
```

Comment:
We should also check to make sure both the texture and the sampler get correct binding index in the output metal source.

Reply:
I will try that tomorrow.